sdk/core/azure-core/src/io/body_stream.cpp

```diff
@@ -89,6 +92,8 @@ int64_t MemoryBodyStream::OnRead(uint8_t* buffer, int64_t count, Context const&
 
 FileBodyStream::FileBodyStream(const std::string& filename)
 {
+  AZURE_ASSERT_MSG(filename.size() > 0, "The file name must not be an empty string.");
```

Comment:
It seems peculiar to me that this particular invalid file name is forbidden by precondition, whereas all other invalid file names result in a runtime error. I'd remove this precondition and let all invalid filenames be treated the same.

Reply:
Unlike other filename checks, the size of string is something the caller can trivially check. At least, that's what motivated this change. An empty string size is potentially an upstream string parsing bug, whereas invalid filename could be caused by file not existing or other reasons, so the type of root causes of the errors seem sufficiently different to me.